### ISSUE TITLE
[todo-mvvm-live-kotlin] Espresso idle counter gets corrupted

### ISSUE THREAD

**Issue by FlexW** [googlesamples/android-architecture#657]
In todo-mvvm-live-kotlin in Util/ViewExt.kt, in the callback of the showSnackbar function there gets the EspressoIdlingResource incremented and decremented.
The problem is, that if onDismissed is called before onShown, the counter gets corrupted. 
This case may happen if showSnackbar is called two times in a really short period of time.

**Comment by JoseAlcerreca** [comment]
Yeah we fixed this in the testing codelab, which will be merged soon with this project.

**Comment by JoseAlcerreca** [comment]
Actually let's keep this one open until we merge.

**Comment by cuong292** [comment]
@JoseAlcerreca hi, can u please provide the reason this branch ( maybe all) using full of activity and inside activity have only 1 fragment ( which i remember that google recommended to use 1 activity 1 fragment )
